broker/api.py

```diff
@@ -278,59 +263,34 @@ def update(
             noop = False
 
             if "origin" in params:
-                if params["origin"] is None:
-                    instance.cloudfront_origin_hostname = (
-                        config.DEFAULT_CLOUDFRONT_ORIGIN
-                    )
-                    # make sure HOST is in forwarded_headers. Do it by messing with params
-                    # to trick the logic below into updating it.
-                    params["forward_headers"] = params.get(
-                        "forward_headers", []
-                    ).append("HOST")
+                if params["origin"]:
+                    origin_hostname = params["origin"]
                 else:
-                    instance.cloudfront_origin_hostname = params["origin"]
+                    origin_hostname = config.DEFAULT_CLOUDFRONT_ORIGIN
+                instance.cloudfront_origin_hostname = origin_hostname
```

Comment:
For this check and the one below, is it only necessary to set the `origin_hostname` or `cloudfront_origin_path` if those keys exist in the `params` in the first place, or should they always have a value?

If it's only if the keys exist then this is fine!  If there must always be something though, could these checks be consolidated slightly, such as:

```python
# params.get() will return None by default if the dict key doesn't exist: https://docs.python.org/3/library/stdtypes.html#dict.get
origin_hostname = params.get('origin')

if origin_hostname is None:
    origin_hostname = config.DEFAULT_CLOUDFRONT_ORIGIN

instance.cloudfront_origin_hostname = origin_hostname
```

Actually, might this be possible anyway even if the initial key checks need to stay as is?

Reply:
yeah, it's only if the key exists. The OSBAPI says we should not change anything that is not in the update parameters, so we leave it as-is otherwise. We then check for the truthiness of the origin. The intent here is that a user could update to not use a custom origin by passing `"origin": null` or `"origin": ""`

Reply:
Ah okay, so leave it alone unless the user has explicitly done something different here.  Thanks!